### PR TITLE
Adds Proguard deobfuscation support.

### PR DESCRIPTION
**Pull request by jnlopar**
This patch has been really useful to me so far in analyzing methods on some of our obfuscated YouTube release builds. It does restructure things a bit, such as making objects for the names and using Args4J for command line processing, so interested to hear if you'd like to merge something like this in. I know the alternative would be to use unobfuscated Proguard builds, but this allows users to analyze exact release builds, which I think adds some value to the app.

Also, very much enjoying getting to play with some Kotlin, so please let me know if my usage of the language is wonky anywhere so I can get to know the idioms better. Thanks!

### CODE REVIEW

File: src/main/kotlin/com/jakewharton/dex/DexMethods.kt [hunk 1]
```diff
@@ -18,20 +21,37 @@ import java.util.zip.ZipInputStream
...
+
+      val proguardMap = config.proguardMap
+      val methodMapper : ((MethodName) -> MethodName) =
+          if (proguardMap != null) ProguardNameMapper(proguardMap) else ({ it })
+
+      val bytesList = config.files
           .map { FileInputStream(it) }
           .defaultIfEmpty(System.`in`)
-          .map { it.readBytes() }
+          .map { it.use { it.readBytes() } }
```
Comment (JakeWharton) [inline src/main/kotlin/com/jakewharton/dex/DexMethods.kt]: Subtle bug fix!

Comment (jnlopar) [inline src/main/kotlin/com/jakewharton/dex/DexMethods.kt]: Thanks :) If you'd like it as a separate PR (since it's much smaller) I can send it out

**Comment by JakeWharton** [comment]
Had/have a busy week. Haven't forgotten about this though!

**Comment by JakeWharton** [comment]
So I finally got around to finishing and sending a PR (#35) which adds a model object for methods. It's not too dissimilar from what you have in here.

I'm going to adapt what you have here to this new model but using a slightly different non-CLI API. I'll commit with you as the author though, since it's just an adaptation of this work.

Just wanted to give you a heads up that I'm (finally) back to working on this and you don't need to bother rebasing.

**Comment by jnlopar** [comment]
Sounds good to me, thanks for taking it on!

**Comment by JakeWharton** [comment]
Well it only took 3 years... sorry for sitting on this for so long. I ended up just rewriting it because the library/tool has changed so significantly.

**Pull request closed without merging**